lib/bson/ext_json.rb

```diff
@@ -219,6 +219,14 @@ module ExtJSON
             raise Error::ExtJSONParseError, "Invalid subType value in $binary: #{value}"
           end
           create_binary(encoded_value, subtype)
+
+        when '$uuid'
+          unless /\A[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}\z/.match(hash['$uuid'])
+            raise Error::ExtJSONParseError, "Invalid $uuid value: #{hash}"
```

Comment:
The value given in the error message should be the value being validated, to be consistent with the other branches. I suggest assigning `hash[$uuid]` to a variable and using it here and in the `from_uuid` call below.